pandas/io/json/_normalize.py

```diff
@@ -531,7 +532,14 @@ def _recursive_extract(data, path, seen_meta, level=0):
             raise ValueError(
                 f"Conflicting metadata name {k}, need distinguishing prefix "
             )
-        result[k] = np.array(v, dtype=object).repeat(lengths)
+        if v and is_list_like(v[0]):
+            out = []
+            for item, repeat in zip(v, lengths):
+                for _ in range(repeat):
+                    out.append(item)
+        else:
+            out = np.array(v, dtype=object).repeat(lengths).tolist()
```

Comment:
Here `dtype=object` is necessary since np.array(["a", np.nan]) will result in the missing value is converted to string format "nan".

Reply:
Is the `tolist` call needed as well?

Reply:
Not necessary, but auto-type-infering won't be triggered if drop this. I think type infer is useful here. If we still want keep the object type, I can change. Suggestion?

Reply:
I was hoping this line could avoid `dtype=object` and `tolist` such that this line doesn't trigger type inference and would be more performant. Otherwise since both `if/else` blocks convert to list are essentially similar.

Reply:
How about the following solution? We only point out the object dtype when this is a string np.array and do the `tolist()` when nested array detected. For the numeric dtype, we can still keep performant.
```python
if v and isinstance(v[0], str):
    out = np.array(v, dtype=object)
else:
    out = np.array(v)

out = out.repeat(lengths, axis=0)
if v and is_list_like(v[0]):
    out = out.tolist()
```

Reply:
Actually looking back at the original line, it appears `dtype=object` was originally here and allowing dtype inference all the time might actually lead to more desired behavior (more specific types).

Would it make sense just to combine both branches then and always return a list (without using `np.array`)? Would be good to also to explore if there's any performance hit